File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/dagger/GrpcCallScopeGraph.java
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.dagger;

import com.geekinasuit.daggergrpc.api.GrpcCallScope;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.services.HelloWorldService;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.services.WhateverService;
import dagger.Subcomponent;
import javax.annotation.processing.Generated;

@Generated("to be generated")
@Subcomponent(modules = {GrpcCallScopeGraphModule.class})
@GrpcCallScope
public interface GrpcCallScopeGraph {
  HelloWorldService helloWorld();

  WhateverService whatever();

  interface Supplier {
    GrpcCallScopeGraph callScope();
  }
}

File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/dagger/GrpcCallScopeGraph.kt
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.dagger

import com.geekinasuit.daggergrpc.api.GrpcCallScope
import javax.annotation.Generated
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.HelloWorldService
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.WhateverService
import dagger.Subcomponent

@Generated
@Subcomponent(modules = [GrpcCallScopeGraphModule::class])
@GrpcCallScope
interface GrpcCallScopeGraph {
  fun helloWorld(): HelloWorldService

  fun whatever(): WhateverService

  interface Supplier {
    fun callScope(): GrpcCallScopeGraph
  }
}

